### PR TITLE
Fix `return_attention_scores` bug

### PR DESCRIPTION
**Pull request by abheesht17**

### CODE REVIEW

File: keras_hub/src/layers/modeling/transformer_encoder.py [hunk 1]
```diff
@@ -207,22 +207,13 @@ def call(
...
-                query=x,
-                value=x,
-                attention_mask=self_attention_mask,
-                training=training,
-            )
+        x, attention_scores = self._self_attention_layer(
+            query=x,
+            value=x,
+            attention_mask=self_attention_mask,
+            return_attention_scores=True,
```
Comment (SamanehSaadat) [inline keras_hub/src/layers/modeling/transformer_encoder.py:214]: I was looking to see how `return_attention_scores` is used in MHA and looking at [this](https://github.com/keras-team/keras/blob/8409e18dd73dcffaa4c5772326ce33866d8c4df5/keras/src/layers/attention/multi_head_attention.py#L433), I was thinking if setting `return_attention_scores=True` here could cause any problems. If we set it `True` here, the user doesn't have a way to disable it if they want to use flash attention, right?

Comment (abheesht17) [reply keras_hub/src/layers/modeling/transformer_encoder.py:214]: Ah, good spot. I'll keep the if...else, then

Comment (SamanehSaadat) [reply keras_hub/src/layers/modeling/transformer_encoder.py]: I think we can still get rid of the `if-else` block by directly passing the flag to MHA (setting it like this `return_attention_scores=return_attention_scores`). We don't have to pass the flag only when it's `True`. wdyt?

Comment (SamanehSaadat) [reply keras_hub/src/layers/modeling/transformer_encoder.py]: You're right! Let's keep it the way it is then!

**Comment by SamanehSaadat** [comment]
@abheesht17 I was wondering if you've checked why one of the tests is failing.

**Comment by SamanehSaadat** [comment]
> @abheesht17 I was wondering if you've checked why one of the tests is failing.

I think this might be the root cause: https://github.com/keras-team/keras/pull/20448/files#r1837037902

**Review by SamanehSaadat — APPROVED** [review]
Thanks, Abheesht, for catching and fixing this bug!

**Comment by SamanehSaadat** [comment]
I'll merge as the test failure is not related to this PR.

**Comment by abheesht17** [comment]
> > @abheesht17 I was wondering if you've checked why one of the tests is failing.
> 
> I think this might be the root cause: https://github.com/keras-team/keras/pull/20448/files#r1837037902

I'll take a look later today! Sorry for the delay

**Comment by abheesht17** [comment]
> > > > @abheesht17 I was wondering if you've checked why one of the tests is failing.
> > > 
> > > 
> > > I think this might be the root cause: https://github.com/keras-team/keras/pull/20448/files#r1837037902
> > 
> > 
> > I'll take a look later today! Sorry for the delay
> 
> All good! The test failure was unrelated to this PR and it should be fixed now ([keras-team/keras#20482](https://github.com/keras-team/keras/pull/20482)).

Awesome!

**Pull request merged**